pkg/apis/maistra/conversion/extension_providers.go

```diff
@@ -18,34 +16,120 @@ func populateExtensionProvidersValues(in *v2.ControlPlaneSpec, values map[string
 		}
 	}
 
+	var extensionProvidersValues []map[string]interface{}
 	for _, ext := range in.ExtensionProviders {
-		if ext.Prometheus == nil {
-			return fmt.Errorf("extension provider entry %s does not define any provider - it must specify one of: prometheus", ext.Name)
-		}
-		prometheus := []map[string]interface{}{
-			{
+		if ext.Prometheus != nil {
+			extensionProvidersValues = append(extensionProvidersValues, map[string]interface{}{
 				"name":       ext.Name,
 				"prometheus": map[string]interface{}{},
-			},
+			})
 		}
-		if err := setHelmMapSliceValue(values, "meshConfig.extensionProviders", prometheus); err != nil {
-			return err
+		if ext.EnvoyExtAuthzHTTP != nil {
+			envoyExtAuthzHTTPValues := map[string]interface{}{
+				"service": ext.EnvoyExtAuthzHTTP.Service,
+				"port":    ext.EnvoyExtAuthzHTTP.Port,
+			}
+			if ext.EnvoyExtAuthzHTTP.Timeout != nil {
+				envoyExtAuthzHTTPValues["timeout"] = *ext.EnvoyExtAuthzHTTP.Timeout
+			}
+			if ext.EnvoyExtAuthzHTTP.PathPrefix != nil {
+				envoyExtAuthzHTTPValues["pathPrefix"] = *ext.EnvoyExtAuthzHTTP.PathPrefix
+			}
+			if ext.EnvoyExtAuthzHTTP.FailOpen != nil {
+				envoyExtAuthzHTTPValues["failOpen"] = *ext.EnvoyExtAuthzHTTP.FailOpen
+			}
+			if ext.EnvoyExtAuthzHTTP.StatusOnError != nil {
+				envoyExtAuthzHTTPValues["statusOnError"] = *ext.EnvoyExtAuthzHTTP.StatusOnError
+			}
+			if ext.EnvoyExtAuthzHTTP.IncludeRequestHeadersInCheck != nil {
+				envoyExtAuthzHTTPValues["includeRequestHeadersInCheck"] = stringToInterfaceArray(ext.EnvoyExtAuthzHTTP.IncludeRequestHeadersInCheck)
+			}
+			if ext.EnvoyExtAuthzHTTP.IncludeAdditionalHeadersInCheck != nil {
+				envoyExtAuthzHTTPValues["includeAdditionalHeadersInCheck"] = mapOfStringToInterface(ext.EnvoyExtAuthzHTTP.IncludeAdditionalHeadersInCheck)
+			}
+			if ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck != nil {
+				includeRequestBodyInCheckValues := map[string]interface{}{}
+				if ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck.MaxRequestBytes != nil {
+					includeRequestBodyInCheckValues["maxRequestBytes"] = *ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck.MaxRequestBytes
+				}
+				if ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck.AllowPartialMessage != nil {
+					includeRequestBodyInCheckValues["allowPartialMessage"] = *ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck.AllowPartialMessage
+				}
+				if ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck.PackAsBytes != nil {
+					includeRequestBodyInCheckValues["packAsBytes"] = *ext.EnvoyExtAuthzHTTP.IncludeRequestBodyInCheck.PackAsBytes
+				}
+				envoyExtAuthzHTTPValues["includeRequestBodyInCheck"] = includeRequestBodyInCheckValues
+			}
+			extensionProvidersValues = append(extensionProvidersValues, map[string]interface{}{
+				"name":              ext.Name,
+				"envoyExtAuthzHttp": envoyExtAuthzHTTPValues,
+			})
```

Comment:
nit: If you extract `ext.EnvoyExtAuthzHTTP` into a variable, a lot of this code becomes shorter.

Reply:
you could also rename `envoyExtAuthzHTTPValues` to `v` or something short like that to make this code a bit faster to read.

Reply:
Same in `populateExtensionProvidersConfig`.

Reply:
ha, yeah, me too. I also used to prefer long variable names, but now I see the benefit of having short names for variables that have limited scope. And with this much code, it makes a real difference if the repetitive variable name is short.

Reply:
Please take a look once again. I renamed variables to be shorter and readable (for ex-Java devs) as well. I also refactored these functions a little bit, so that there is less conditions and indentations.

Reply:
I like the `allValues` & `values` solution. And using `provider` instead of `ext`. Can you also rename `includeRequestBodyInCheckValues` to something shorter?